From e98a7835708a6f987573bef30570cff8698a9217 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 10 Nov 2017 00:16:48 +0800
Subject: [PATCH 2772/5242] MLK-16775-2 usb: xhci: plat: only do high bus freq
 for runtime pm

commit  3ae6f3e9217310c2488836e010eff25d8afeaca0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Remove xhci_suspend and xhci_resume as i.MX8MQ dwc3 can't support it,
add high bus request and release, and enable runtime pm by default.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/host/xhci-plat.c |   23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index b4d576a..0500af6 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -18,6 +18,7 @@
 #include <linux/usb/phy.h>
 #include <linux/slab.h>
 #include <linux/acpi.h>
+#include <linux/busfreq-imx.h>
 
 #include "xhci.h"
 #include "xhci-plat.h"
@@ -304,6 +305,8 @@ static int xhci_plat_probe(struct platform_device *pdev)
 		hcd->skip_phy_initialization = 1;
 	}
 
+	request_bus_freq(BUS_FREQ_HIGH);
+
 	ret = usb_add_hcd(hcd, irq, IRQF_SHARED);
 	if (ret)
 		goto disable_usb_phy;
@@ -318,12 +321,6 @@ static int xhci_plat_probe(struct platform_device *pdev)
 	device_enable_async_suspend(&pdev->dev);
 	pm_runtime_put_noidle(&pdev->dev);
 
-	/*
-	 * Prevent runtime pm from being on as default, users should enable
-	 * runtime pm using power/control in sysfs.
-	 */
-	pm_runtime_forbid(&pdev->dev);
-
 	return 0;
 
 
@@ -332,6 +329,7 @@ static int xhci_plat_probe(struct platform_device *pdev)
 
 disable_usb_phy:
 	usb_phy_shutdown(hcd->usb_phy);
+	release_bus_freq(BUS_FREQ_HIGH);
 
 put_usb3_hcd:
 	usb_put_hcd(xhci->shared_hcd);
@@ -375,6 +373,7 @@ static int xhci_plat_remove(struct platform_device *dev)
 
 	pm_runtime_set_suspended(&dev->dev);
 	pm_runtime_disable(&dev->dev);
+	release_bus_freq(BUS_FREQ_HIGH);
 
 	return 0;
 }
@@ -410,18 +409,14 @@ static int __maybe_unused xhci_plat_resume(struct device *dev)
 
 static int __maybe_unused xhci_plat_runtime_suspend(struct device *dev)
 {
-	struct usb_hcd  *hcd = dev_get_drvdata(dev);
-	struct xhci_hcd *xhci = hcd_to_xhci(hcd);
-
-	return xhci_suspend(xhci, true);
+	release_bus_freq(BUS_FREQ_HIGH);
+	return 0;
 }
 
 static int __maybe_unused xhci_plat_runtime_resume(struct device *dev)
 {
-	struct usb_hcd  *hcd = dev_get_drvdata(dev);
-	struct xhci_hcd *xhci = hcd_to_xhci(hcd);
-
-	return xhci_resume(xhci, 0);
+	request_bus_freq(BUS_FREQ_HIGH);
+	return 0;
 }
 
 static const struct dev_pm_ops xhci_plat_pm_ops = {
-- 
1.7.9.5

